src/Microsoft.PowerShell.Commands.Utility/commands/utility/XmlCommands.cs

```diff
@@ -437,31 +440,40 @@ protected override void BeginProcessing()
         /// </summary>
         protected override void ProcessRecord()
         {
-            if (As.Equals("Stream", StringComparison.OrdinalIgnoreCase))
+            switch(Format)
             {
-                CreateMemoryStream();
+                case "CliXml":
+                    SerializeObject();
+                    break;
 
-                if (_serializer != null)
-                    _serializer.SerializeAsStream(InputObject);
+                default:
+                    if (As.Equals("Stream", StringComparison.OrdinalIgnoreCase))
+                    {
+                        CreateMemoryStream();
 
-                if (_serializer != null)
-                {
-                    _serializer.DoneAsStream();
-                    _serializer = null;
-                }
-                // Loading to the XML Document
-                _ms.Position = 0;
-                StreamReader read = new StreamReader(_ms);
-                string data = read.ReadToEnd();
-                WriteObject(data);
+                        if (_serializer != null)
+                            _serializer.SerializeAsStream(InputObject);
 
-                // Cleanup
-                CleanUp();
-            }
-            else
-            {
-                if (_serializer != null)
-                    _serializer.Serialize(InputObject);
+                        if (_serializer != null)
+                        {
+                            _serializer.DoneAsStream();
+                            _serializer = null;
+                        }
+                        // Loading to the XML Document
+                        _ms.Position = 0;
+                        StreamReader read = new StreamReader(_ms);
```

Comment:
DRY - but I'd recommend a slightly more self-explanatory variable name
```suggestion
                        var reader = new StreamReader(_ms);
```